token-swap/program/src/processor.rs

```diff
@@ -357,6 +393,8 @@ impl Processor {
         let destination_info = next_account_info(account_info_iter)?;
         let pool_mint_info = next_account_info(account_info_iter)?;
         let pool_fee_account_info = next_account_info(account_info_iter)?;
+        let source_token_mint_info = next_account_info(account_info_iter)?;
+        let destination_token_mint_info = next_account_info(account_info_iter)?;
```

Comment:
ill review the rest of the pr in more detail but i think im ok with this. the main thing that bothered me was the addition of the mints without checking their validity, but i see they flow through `transfer_checked` in all cases. still it might be nice to compare their addresses to the mints on the token accounts after parsing those on line 445, for redundancy

Reply:
Exactly, the idea is to let the token program perform the mint checks.  If we were optimizing this program for compute units, like Token (sometimes), I would keep the checks out, but since this is a reference implementation, I'm happy to add them in.